viewer/src/public/migration/Cognite3DViewer.ts

```diff
@@ -148,6 +152,14 @@ export class Cognite3DViewer {
     this.revealManager = new RevealManagerBase(this.cadManager, this.materialManager, this.pointCloudManager);
     this.startPointerEventListeners();
 
+    this.sectorRepository.getLoadingStateObserver().subscribe(isLoading => {
+      if (isLoading) {
+        this.spinner.show();
+      } else {
+        this.spinner.hide();
+      }
+    });
```

Comment:
Not a biggie, but I think we should avoid RxJS in the 'migration'-layer.

Reply:
Well, it's used already anyway. It's not the first usage. 

```js
    this.sectorRepository
      .getParsedData()
      .pipe(
        share(),
        filter(x => x.blobUrl === cadNode.cadModelMetadata.blobUrl)
      )
      .subscribe(parseSector => model3d.updateNodeIdMaps(parseSector));
```

TBH, I think it's not easy to avoid it because the `CachedRepository` is built on top of it. As I said before: rxjs code always tends to spread. It's just so convenient to use it for that loader case :D 

To avoid it here I will probably need to do exactly the same thing I suggested for the reveal API. Some `.on/.off` to watch for loading state changes. But it feels like way more code.

Reply:
Yeah I agree with Maksim here. adding a single observer which you observe is not a very complex and it doesn't leak outside the migration layer.

Reply:
This can also be wrapped in the RevealManager as a event function like suggested earlier.

```revealManager.on('loadingStateChanged', isLoading => {
  // Do own loading logic
});```

Reply:
@larsmoa @j-bjorne so just to summarize both threads. Are we agreed on:

1. I should change the RevealManager public API like that

```diff
const loadingHandler = (isLoading) => /* custom loading handling */
+ revealManager.on('loadingStateChanged', loadingHandler);
- revealManager.getLoadingStateObserver().subscribe(loadingHandler );
```

2. Cognite3DViewer is ok now. Since rxjs is internal and is not exposed to public API.

?

Reply:
I agree with both of those

Reply:
Agreed, @maksnester